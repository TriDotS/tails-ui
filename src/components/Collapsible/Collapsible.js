import React, { PureComponent } from 'react'

import PropTypes from 'prop-types'

export default class Collapsible extends PureComponent {
  static propTypes = {
    color: PropTypes.string,
    title: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired
  }

  static defaultProps = {
    color: 'grey'
  }

  state = {
    collapsed: false
  }

  handleToggle = () => {
    this.setState({
      collapsed: !this.state.collapsed
    })
  }

  render() {
    const { color, title, children } = this.props
    const { collapsed } = this.state

    return (
      <div className={`border-2 border-${color}-light rounded`}>
        <div
          className={`flex flex-row items-center justify-between w-full border-b-2 border-${color}-light p-3 `}
        >
          <div className="text-2xl text-grey-darker">{title}</div>
          <div onClick={this.handleToggle} className="cursor-pointer">
            {!collapsed && (
              <svg width="16" height="16" viewBox="0 0 32 32" fill="#606f7b">
                <path d="M32 16c0-8.837-7.163-16-16-16s-16 7.163-16 16 7.163 16 16 16 16-7.163 16-16zM3 16c0-7.18 5.82-13 13-13s13 5.82 13 13-5.82 13-13 13-13-5.82-13-13z" />
                <path d="M9.914 11.086l-2.829 2.829 8.914 8.914 8.914-8.914-2.828-2.828-6.086 6.086z" />
              </svg>
            )}
            {collapsed && (
              <svg width="16" height="16" viewBox="0 0 32 32" fill="#606f7b">
                <path d="M0 16c0 8.837 7.163 16 16 16s16-7.163 16-16-7.163-16-16-16-16 7.163-16 16zM29 16c0 7.18-5.82 13-13 13s-13-5.82-13-13 5.82-13 13-13 13 5.82 13 13z" />
                <path d="M22.086 20.914l2.829-2.829-8.914-8.914-8.914 8.914 2.828 2.828 6.086-6.086z" />
              </svg>
            )}
          </div>
        </div>
        {!collapsed && <div className="p-3">{children}</div>}
      </div>
    )
  }
}
