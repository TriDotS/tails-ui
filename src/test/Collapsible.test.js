import Collapsible from '../components/Collapsible'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <Collapsible title="Collapsible">
      <div>Some content</div>
    </Collapsible>,
    div
  )
})

test('collapsible snapshot', () => {
  const tree = renderer
    .create(
      <Collapsible title="Collapsible">
        <div>Some content</div>
      </Collapsible>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
