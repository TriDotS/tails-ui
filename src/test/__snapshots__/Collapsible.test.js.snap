// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapsible snapshot 1`] = `
<div
  className="border-2 border-grey-light rounded"
>
  <div
    className="flex flex-row items-center justify-between w-full border-b-2 border-grey-light p-3 "
  >
    <div
      className="text-2xl text-grey-darker"
    >
      Collapsible
    </div>
    <div
      className="cursor-pointer"
      onClick={[Function]}
    >
      <svg
        fill="#606f7b"
        height="16"
        viewBox="0 0 32 32"
        width="16"
      >
        <path
          d="M32 16c0-8.837-7.163-16-16-16s-16 7.163-16 16 7.163 16 16 16 16-7.163 16-16zM3 16c0-7.18 5.82-13 13-13s13 5.82 13 13-5.82 13-13 13-13-5.82-13-13z"
        />
        <path
          d="M9.914 11.086l-2.829 2.829 8.914 8.914 8.914-8.914-2.828-2.828-6.086 6.086z"
        />
      </svg>
    </div>
  </div>
  <div
    className="p-3"
  >
    <div>
      Some content
    </div>
  </div>
</div>
`;
